Animations/Animation.js

```diff
@@ -37,20 +22,30 @@ var makeTransition = function (displayName, transitions) {
       // componentDidEnter FUNCTION: use to define actions to run after an animation has entry completed
       // componentDidLeave FUNCTION: use to define actions to run after an animation has leave completed
   return React.createClass({
-    componentWillEnter: function (callback) {
-      setAnimationDomNode('enter', callback);
+    // setAnimationDomNode FUNCTION: function to handle manipulating and setting a TransitionGroup in the DOM
+      //@param action STRING: the transition key you are looking to affect
+      //@param callback FUNCTION: callback attribute passed in from the containing function to act on function completion
+    setAnimationDomNode: function (action, callback) {
+      this.el = this.getDOMNode();
+      // requestAnimationFrame FUNCTION: Calls the specified function updating an animation before the next browser repaint. Defined in window
+      window.requestAnimationFrame(function () {
+        for (var key in transitions[action]) {
```

Comment:
Store `transitions[actions]` in a variable so you do not need to keep looking it up.